internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -3285,6 +3331,33 @@ func expandKubernetesClusterAutoScalerProfile(input []interface{}) *managedclust
 	}
 }
 
+func expandKubernetesClusterAzureKeyVaultKms(d *pluginsdk.ResourceData, input []interface{}) (*managedclusters.AzureKeyVaultKms, error) {
+	if ((input == nil) || len(input) == 0) && d.HasChanges("key_vault_kms") {
+		return &managedclusters.AzureKeyVaultKms{
+			Enabled: utils.Bool(false),
+		}, nil
+	} else if (input == nil) || len(input) == 0 {
+		return nil, nil
+	}
+
+	raw := input[0].(map[string]interface{})
+	kvAccess := managedclusters.KeyVaultNetworkAccessTypes(raw["key_vault_network_access"].(string))
+
+	azureKeyVaultKms := &managedclusters.AzureKeyVaultKms{
+		Enabled:               utils.Bool(raw["enabled"].(bool)),
+		KeyId:                 utils.String(raw["key_id"].(string)),
+		KeyVaultNetworkAccess: &kvAccess,
+		KeyVaultResourceId:    utils.String(raw["key_vault_resource_id"].(string)),
+	}
+
+	private := managedclusters.KeyVaultNetworkAccessTypesPrivate
+	if kvAccess == private && len(*azureKeyVaultKms.KeyVaultResourceId) == 0 {
```

Comment:
If `azureKeyVaultKms.KeyVaultResourceId` is `nil`, then the dereference here would cause a panic.
If not, then what we're checking here is `*azureKeyVaultKms.KeyVaultResourceId == ""`